src/notification.c

```diff
@@ -649,29 +649,39 @@ void notification_update_text_to_render(struct notification *n)
 }
 
 /* see notification.h */
-void notification_do_action(const struct notification *n)
+void notification_do_action(struct notification *n)
 {
+        assert(n->default_action_name);
+        
         if (g_hash_table_size(n->actions)) {
-                if (g_hash_table_contains(n->actions, "default")) {
-                        signal_action_invoked(n, "default");
+                if (g_hash_table_contains(n->actions, n->default_action_name)) {
+                        signal_action_invoked(n, n->default_action_name);
                         return;
                 }
-                if (g_hash_table_size(n->actions) == 1) {
```

Comment:
I was thinking we'd keep this a well. It may be useful and it's more backward compatible. You'll only need to do it when the action name is default.

Reply:
You're right, I forgot that when `action_name` is default this is actually intended behavior.